BedrockServer.cpp

```diff
@@ -1952,11 +1952,23 @@ void BedrockServer::_prePollCommands(fd_map& fdm) {
     for (auto& command : _outstandingHTTPSCommands) {
         command->prePoll(fdm);
     }
+
+    // Make sure that waiting for an HTTPS command interrupts the current `poll` in the sync thread.
+    _newCommandsWaiting.prePoll(fdm);
```

Comment:
I'm not really following now this interrupts `poll` (which I assume is the `S_poll` call here) https://github.com/Expensify/Bedrock/blob/aff166967baeb439d561a5e12a46742573d1f40e/BedrockServer.cpp#L322

Though that's probably because I only just tried to read what `poll` does https://man7.org/linux/man-pages/man2/poll.2.html. 

So does `poll` basically act like a `while()` statement where it waits for a file descriptor (which could be a socket/thread?) to signal "hey i'm done" or "i'm ready"? 

So I guess that means this interrupts `poll` because calling `SSynchronizedQueue::prePoll` will reset the file descriptor back to `SREADEVTS` which are "flags" that signify there is data to be read and `poll` will exit/continue?

I see `SREADEVTS` is actually one of these 3 https://github.com/Expensify/Bedrock/blob/aff16696/libstuff/libstuff.h#L495

that linux man page shows:
<img width="649" alt="image" src="https://user-images.githubusercontent.com/4073354/160012965-7796e468-e23e-4571-993c-e6435127f5b4.png">
<img width="651" alt="image" src="https://user-images.githubusercontent.com/4073354/160013077-9090f573-d961-4a5b-b9f1-133f8ca17194.png">

I don't really follow the middle one but I suspect we mostly care about `POLLIN` and `POLLUP` which I believe are signaling that the fd is ready or the other end disconnected and thus we can stop waiting?

Reply:
`poll` takes a list of file descriptors and a set of events. It then blocks until at least one of those events is ready on at least one of those file descriptors.

Conceptually, it's like this:

```
list<FDs> someFiles = getSomeFiles();
list<EVENTS> someEvents = justReadEvents();

poll(someFiles, someEvents);
// poll blocks until one of the files in someFiles is readable (i.e., it has received some data we can read).
```

There's no loop, poll simply runs until one of these files is ready to read.